test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/security/SecureRandomTest.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * scalajs-java-securerandom (https://github.com/scala-js/scala-js-java-securerandom)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.security
+
+import org.junit.Test
+import org.junit.Assert._
+import org.junit.Assume._
+
+import java.security.SecureRandom
+
+import org.scalajs.testsuite.utils.BuildInfo
+
+class SecureRandomTest {
+  @Test def nextBytes(): Unit = {
+    assumeFalse("expecting failure", BuildInfo.expectFailure)
+
+    val random = new SecureRandom()
+
+    val buffer = new Array[Byte](24)
+    assertTrue("confidence check", buffer.count(_ == 0) == 24)
+
+    random.nextBytes(buffer)
+
+    // the likelihood of this failing should be one in 4 billion
+    assertTrue("nextBytes() must fill the provided buffer", buffer.count(_ == 0) < 4)
+
+    val buffer2 = new Array[Byte](24)
+    random.nextBytes(buffer2)
+    assertFalse("nextBytes() must return different results", buffer2.sameElements(buffer))
+  }
+
+  @Test def nextInt(): Unit = {
+    assumeFalse("expecting failure", BuildInfo.expectFailure)
+
+    val random = new SecureRandom()
+
+    val i1 = random.nextInt()
+    val i2 = random.nextInt()
+    assertNotEquals(i2, i1)
+  }
+
+  @Test def nextFloat(): Unit = {
+    assumeFalse("expecting failure", BuildInfo.expectFailure)
+
+    /* `nextFloat()` actually relies on `next(n)` clearing the higher order
+     * bits for the resulting float not to exceed 1.0f, so this test is an
+     * indirect way of testing that specification.
+     */
+
+    val random = new SecureRandom()
+    for (_ <- 0 until 100) {
+      val f = random.nextFloat()
+      assertTrue(f >= 0.0f && f < 1.0f)
+    }
+  }
+
+  @Test def verifyFailure(): Unit = {
```

Comment:
use `@Test(expect = classOf[UnsupportedOperationException)`? (also in UUIDTest)

Reply:
That doesn't appear to work in this case. When the assumption fails, it fails with
```
Unexpected exception, expected<java.lang.UnsupportedOperationException> but was<org.junit.AssumptionViolatedException>
```
So we have to keep the manual thing.